functions/callback-functions.md

```diff
@@ -0,0 +1,214 @@
+  cleanupBehavior();
+}
+```
+<!-- prettier-ignore-end -->
+
+We call `cleanupBehavior` a _callback function_ inside `bakeCake`. It refers to the expectation that `bakeCake` should _call back_ a function... the function that was passed in, when `bakeCake` was invoked in the first place.
+
+## Revisiting Calculator
+
+Let's consider how we can build our calculator function using callbacks. This process will be very similar to our `bakeCake` example, but we'll be mindful about how arguments are passed in.
+
+Consider this code:
+
+<!-- prettier-ignore-start -->
+```js
+const multiply = function (factor_a, factor_b) {
+  console.log('Now we\'re in multiply~~');
+  return factor_a * factor_b;
+}
+
+const calculate = function (a, b, operate) {
+  console.log('We\'re inside the calculate function!');
+  return operate(a, b);
+}
+
+const result = calculate(7, 6, multiply);
+console.log('Our final result from calculate is:', result);
+```
+<!-- prettier-ignore-end -->
+
+After running this code, we'll see this output:
+
+```
+We're inside the calculate function!
+Now we're in multiply~~
+Our final result from calculate is: 42
+```
+
+### Practice Tracing Code
+
+Let's practice tracing code, going through it and following the flow of execution. The first several lines of this snippet define functions. Our first moment of action is the line `const result = calculate(7, 6, multiply);`.
+
+We invoke our `calculate` function when we call it with parentheses `()`, in the expression `calculate(7, 6, multiply)`. Here, we're passing in the `multiply` function, without invoking it.
+
+When we enter our `calculate` function, we have these values:
+
+| Local variable | Value      |
+| -------------- | ---------- |
+| `a`            | `7`        |
+| `b`            | `6`        |
+| `operate`      | `multiply` |
+
+Our `calculate` function eventually invokes `operate` with the expression `operate(a, b)`. When we run `operate(a, b)`, we're invoking the `multiply` function with those arguments.
+
+Inside the `multiply` function, we have these values:
+
+| Local variable | Value |
+| -------------- | ----- |
+| `factor_a`     | `7`   |
+| `factor_b`     | `6`   |
+
+Ultimately, our `multiply` function returns `42`...
+
+<!-- prettier-ignore-start -->
+```js
+return factor_a * factor_b; // 42
+```
+<!-- prettier-ignore-end -->
+
+Which in turn, after we get back to our `calculate` function...
+
+<!-- prettier-ignore-start -->
+```js
+return operate(a, b); // 42
+```
+<!-- prettier-ignore-end -->
+
+Returns `42` itself.
+
+This comes all the way back to our line where we originally started all of this!
+
+<!-- prettier-ignore-start -->
+```js
+const result = calculate(7, 6, multiply);
+```
+<!-- prettier-ignore-end -->
+
+We finally get a value for our variable `result`, which is `42`.
+
+### !callout-info
+
+## Compare to Python
+
+Python actually supports passing around functions, too! Python uses _lambdas_ as a popular pattern for passing around functions.
```

Comment:
I'd leave off the Python comparison, especially to the Python `lambda` keyword. `lambda` might be reasonable to reference in a discussion of anonymous functions, but it's really limited in terms of what it can do. To be as flexible as functions being passed around in JS, it's generally more effective to use named functions in Python. Since I'm assuming we don't want to go into all of this right here, that's why I suggest cutting it.

BUT! We could still end off with a callout that encourages the reader to trace through the `add` scenario.
```suggestion
## Let's Trace the Second Call on Our Own

Our example goes on to call `calculate` with the `add` function. We should trace through this call as well to practice thinking about callbacks!
```